### DISCUSSION TITLE
Any chance for Android availability?

### DISCUSSION THREAD

**Discussion by sparra98** [vkbo/novelWriter#1307] [Q&A]
Is there any chance we can get this on Android (tablets)?

-Thank you

**Comment by vkbo** [comment]
In principle it should be possible. I don't have the capacity to do this, but if someone wants to try it, then I'm all for it.

**Reply by HeyMyian** [reply]
I'm tentatively interested in this. I've never worked with Qt before so I don't at all expect anything useful to come out of my try, but maybe learning something new in the process is the goal.

I had a look around Qt Creator and finished one simple tutorial app on a mobile emulator. When I looked further, I did not see any option to do anything with the novelWriter files. Please forgive my ignorance. Would we need certain novelWriter Qt source files for this? I'd also understand if you don't want to entrust this with a novice. I just wanted to see and understand the complexity of the task.

**Reply by vkbo** [reply]
I did not build novelWriter with QtCreator or any of their GUI design tools. The novelWriter source just imports the PyQt5 wrapper library and builds the GUI directly in the code itself. As long as it can import what it needs, it should run.

What you'd need is a working Python environment with the Qt framework for Android. I know absolutely nothing about Android app development, so I have no idea how an app is packaged. If Qt and Python are running properly, there should be no reason I can think of why novelWriter shouldn't run from source.

That said, novelWriter does depend on lxml to run at the moment. I plan to drop that dependency some time after Python 3.7 has reached end of life (end of June this year). Then the standard Python xml package should be able to replace it. See issue #1257 for more details.

The novelWriter GUI is designed to work with a mouse or through keyboard shortcuts, so I expect it will need some tweaking to be able to work well on a touch screen, but I don't think it will be a major issue.

**Reply by HeyMyian** [reply]
👍 If you're saying I've got all the novelWriter files I need, then that already helps as I don't have to guess anymore. I'll take another shot at figuring something out. Thank you.

**Comment by snden** [comment]
When I review and edit larger passages of a novel, I use an Android e-book reader. I export the entire novel in Novelwriter to a .nwd file, including notes, etc., and edit it as an .md file. After proofreading, I load it back into Novelwriter and divide it again into individual chapters. Thanks for these two export and import options, without them such work would not be possible.

For such situations, but perhaps also for direct writing on Android devices (while traveling, on the bus, train ...), it would be great if there was an Android app.
It could be completely simple, mainly to be able to work with the entire Novelwriter project.
Then it would be enough to simply overwrite the entire project folder from one device to another. Or using special .zip export and import option between devices.

**Reply by vkbo** [reply]
The zip format is coming actually. I'm adding a feature in novelWriter to read and write projects as zip files, just like most office applications do. I will make it optional though.

As for Android support, I have no idea how to do this or how much work it is, or even if the libraries are available in a way that is directly usable. The lxml dependency, which may be problematic, has been removed in the dev branch, so it will be officially gone in novelWriter 2.1. Since the spell check library is optional, the hard requirements are then just Qt5.10+ and Python 3,8+.

As mentioned, I don't have the capacity to investigate, create and maintain an Android release. This will have to be done by a contributor at this point.
